lib/surrealist/value_assigner.rb

```diff
@@ -34,10 +34,23 @@ def assign(schema, instance)
       #
       # @return [Object] value to be further processed
       def raw_value(instance, schema)
-        value = instance.is_a?(Hash) ? instance[schema.key] : instance.send(schema.key)
+        value = instance.is_a?(Hash) ? instance[schema.key] : invoke_method(instance, schema.key)
         coerce_value(value, schema)
       end
 
+      # Checks if there is a custom serializer defined for the object and invokes the method
+      #   on it first.
+      #
+      # @param [Object] instance an instance of a model or a serializer
+      # @param [Symbol] method the schema key that represents the method to be invoked
+      #
+      # @return [Object] the return value of the method
+      def invoke_method(instance, method)
```

Comment:
Looks good but would this make more sense being inside the `Serializer` class?

Reply:
I guess it would, but it's quite complicated to move this method there - it would result in a massive code duplication.

Reply:
Fooling around a bit with the code I see what you mean.